flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java

```diff
@@ -220,7 +220,7 @@
                     .withDescription(
                             Description.builder()
                                     .text(
-                                            "The cluster-id, which should be no more than 45 characters, is used for identifying a unique Flink cluster. "
+                                            "The cluster-id, which should be no more than 40 characters, is used for identifying a unique Flink cluster. "
```

Comment:
Is there anything preventing us from using `Constants.MAXIMUM_CHARACTERS_OF_CLUSTER_ID` here? 🤔

Reply:
@alpreu I'm just puzzled. Why did we change the limit to `40` initially? The current version of the PR sets the limit to `45` again? I double-checked the code that we didn't miss any other label and couldn't find anything else. But it looks like I'm missing something. I remember that the initial change was valid in terms of the number but we just wanted to make it dynamic. Do you see what I'm missing here, @alpreu ? 🤔

Reply:
Yes, it changed back to the original value. The reason is that the 40 was derived from the `"-resourcemanager-leader"` in `KubernetesHaServices`. In that class only configmap names are generated, for which the 63 character limit does not apply. I don't know if it makes sense to move this generation also into `KubernetesLabel`.  Especially because the JobManager one does contain the `JobID` which would restrict the clusterId length extremely.